src/objective_types.jl

```diff
@@ -12,17 +14,26 @@ function fix_order(storage_input, x_input, fun!, fun!_msg)
     end
 end
 # Used for objectives and solvers where no gradient is available/exists
-type NonDifferentiable{T} <: AbstractObjective
+type NonDifferentiable{T<:Real} <: AbstractObjective
```

Comment:
Nothing, it's just a type assertion because it does not make sense for T to be anything else